code/data_structures/src/list/singly_linked_list/menu_interface/singly_list.c

```diff
@@ -0,0 +1,150 @@
+/* This project aims to demonstrate various operations on linked lists. It uses
+a clean command line interface and provides a baseline to make linked list
+applications in C. There are two files. The C file containing main() function and
+invoking functions. The link.h header file where all functions are implemented.
+*/
+
+#include <unistd.h>     // for sleep() function
+#include "link.h"       // custom header file to implement functions
+
+void print();
+void clear();
+int menu();
+
+int main()
+{
+	int c;
+	printf("WELCOME TO THE LINKED LIST APPLICATION\n");
+	clear();
+	usleep(500000);
+
+	do
+	{
+		c = menu();
+	}while(c!=17);
+}
+
+void print()
+{
+	printf("1. Insert at beginning\n");
+	printf("2. Insert at end\n");
+	printf("3. Insert at a particular location\n");
+	printf("4. Insert after a particular value\n");
+	printf("5. Delete from beginning\n");
+	printf("6. Delete from end\n");
+	printf("7. Delete a particular node\n");
+	printf("8. Delete a particular value\n");
+	printf("9. Replace a value at a particular node\n");
+	printf("10. Replace a particular value in the list\n");
+	printf("11. Delete all duplicate values and sort\n");
+	printf("12. Display\n");
+	printf("13. Reverse\n");
+	printf("14. Sort(Ascending)\n");
+	printf("15. Write to File\n");
+	printf("16. Read from a file\n");
+	printf("17. Quit\n");
+}
+
+int menu()
+{
+	int n,a,b;
```

Comment:
Please declare these variables closer to where they are used.

Reply:
These variables are used in that function only hence I think they were positioned right. But I have moved the declaration slightly below.